src/test/java/org/littleshoot/proxy/DirectRequestTest.java

```diff
@@ -0,0 +1,108 @@
+package org.littleshoot.proxy;
+
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.Assert.assertThat;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+
+import java.net.Socket;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.littleshoot.proxy.impl.DefaultHttpProxyServer;
+import org.littleshoot.proxy.test.SocketClientUtil;
+
+/**
+ * This class tests direct requests to the proxy server, which causes endless
+ * loops (#205).
+ */
+public class DirectRequestTest {
+    private HttpProxyServer proxyServer;
+
+    private Socket socket;
+
+    @Before
+    public void setUp() throws Exception {
+        socket = null;
+        proxyServer = null;
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        try {
+            if (proxyServer != null) {
+                proxyServer.abort();
+            }
+        } finally {
+            if (socket != null) {
+                socket.close();
+            }
+        }
+    }
+
+    @Test(timeout = 5000)
+    public void testDirectRequestAnsweredBadRequest() throws Exception {
+
+        HttpFiltersSource filtersSource = new HttpFiltersSourceAdapter() {
+            @Override
+            public HttpFilters filterRequest(HttpRequest originalRequest) {
+                return new HttpFiltersAdapter(originalRequest);
+            }
+        };
+
+        proxyServer = DefaultHttpProxyServer.bootstrap()//
+                .withPort(0)//
+                .withFiltersSource(filtersSource)//
+                .start();
+
+        socket = SocketClientUtil.getSocketToProxyServer(proxyServer);
```

Comment:
I've copied your KeepAliveTest since I'm unhappy with the inherited 
tests in LittleProxy and with the Apache client (deprecated) too. Do you 
see a problem with the socket Client?

I would prefer an URLConnection based HTTP/HTTPS client, since this is 
the regular way in Java/Android to create a client with HTTPS security 
enabled.

Am 19.07.2015 um 21:54 schrieb Jason Hoetger:
> In src/test/java/org/littleshoot/proxy/DirectRequestTest.java
> <https://github.com/adamfisk/LittleProxy/pull/220#discussion_r34961460>:
>
>> +    @Test(timeout = 5000)
>> +    public void testDirectRequestAnsweredBadRequest() throws Exception {
>> +
>> +        HttpFiltersSource filtersSource = new HttpFiltersSourceAdapter() {
>> +            @Override
>> +            public HttpFilters filterRequest(HttpRequest originalRequest) {
>> +                return new HttpFiltersAdapter(originalRequest);
>> +            }
>> +        };
>> +
>> +        proxyServer = DefaultHttpProxyServer.bootstrap()//
>> +                .withPort(0)//
>> +                .withFiltersSource(filtersSource)//
>> +                .start();
>> +
>> +        socket = SocketClientUtil.getSocketToProxyServer(proxyServer);
>
> Just curious why you chose to use the socket-based HTTP client for these
> tests. Would the Apache HTTP Client we use in most other tests not work
> in this case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adamfisk/LittleProxy/pull/220/files#r34961460>.
>